### PR TITLE
[MetricKit] Add support for Xcode 13 beta4.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/metrickit.cs [hunk 1]
```diff
@@ -407,10 +434,13 @@ interface MXMetricManager {
 
 	interface IMXMetricManagerSubscriber { }
 
-	[NoWatch, NoTV, NoMac, iOS (13,0)]
+	[NoWatch, NoTV, Mac (12,0), iOS (13,0)]
 	[Protocol]
 	interface MXMetricManagerSubscriber {
+#if !XAMCORE_4
```
Comment (rolfbjarne) [inline src/metrickit.cs:440]:
```suggestion
#if !XAMCORE_4_0
```

File: tests/xtro-sharpie/MacCatalyst-MetricKit.ignore [hunk 2]
```diff
@@ -1,2 +1,2 @@
+# Added for xamcore
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/MacCatalyst-MetricKit.ignore:1]:
```suggestion
# Added for xamcore 4.0
```

File: tests/xtro-sharpie/iOS-MetricKit.ignore [hunk 3]
```diff
@@ -1,2 +1,2 @@
+# Added for xamcore
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/iOS-MetricKit.ignore:1]:
```suggestion
# Added for xamcore 4.0
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/metrickit.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/metrickit.cs:440]: I'm stupid

File: src/metrickit.cs [hunk 4]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
 		[Export ("timeStampEnd", ArgumentSemantic.Strong)]
 		NSDate TimeStampEnd { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:324]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 5]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("cpuMetrics", ArgumentSemantic.Strong)]
 		MXCpuMetric CpuMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:328]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 6]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("gpuMetrics", ArgumentSemantic.Strong)]
 		MXGpuMetric GpuMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:332]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 7]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("cellularConditionMetrics", ArgumentSemantic.Strong)]
 		MXCellularConditionMetric CellularConditionMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:336]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 8]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("applicationTimeMetrics", ArgumentSemantic.Strong)]
 		MXAppRunTimeMetric ApplicationTimeMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:340]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 9]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("locationActivityMetrics", ArgumentSemantic.Strong)]
 		MXLocationActivityMetric LocationActivityMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:344]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 10]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("networkTransferMetrics", ArgumentSemantic.Strong)]
 		MXNetworkTransferMetric NetworkTransferMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:348]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 11]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("applicationLaunchMetrics", ArgumentSemantic.Strong)]
 		MXAppLaunchMetric ApplicationLaunchMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:352]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 12]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("applicationResponsivenessMetrics", ArgumentSemantic.Strong)]
 		MXAppResponsivenessMetric ApplicationResponsivenessMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:356]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 13]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("diskIOMetrics", ArgumentSemantic.Strong)]
 		MXDiskIOMetric DiskIOMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:360]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 14]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("memoryMetrics", ArgumentSemantic.Strong)]
 		MXMemoryMetric MemoryMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:364]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 15]
```diff
@@ -311,39 +321,51 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[NullAllowed, Export ("displayMetrics", ArgumentSemantic.Strong)]
 		MXDisplayMetric DisplayMetrics { get; }
 
+		[NoMac]
```
Comment (spouliot) [inline src/metrickit.cs:368]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 16]
```diff
@@ -352,7 +374,7 @@ interface MXMetricPayload : NSSecureCoding {
 
 		[Export ("JSONRepresentation")]
 		NSData JsonRepresentation { get; }
-
+#if !MONOMAC
```
Comment (spouliot) [inline src/metrickit.cs:377]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 17]
```diff
@@ -363,24 +385,29 @@ interface MXMetricPayload : NSSecureCoding {
 		[MacCatalyst (14,0)]
 		[Export ("dictionaryRepresentation")]
 		NSDictionary _DictionaryRepresentation14 { get; }
+#else
+		[Export ("dictionaryRepresentation")]
+		NSDictionary DictionaryRepresentation { get; }
+#endif
 
-		[iOS (14,0)]
+		[NoMac, iOS (14,0)]
```
Comment (spouliot) [inline src/metrickit.cs:393]: minor: not needed, there's a `[NoMac]` on the type-level

File: src/metrickit.cs [hunk 18]
```diff
@@ -363,24 +385,29 @@ interface MXMetricPayload : NSSecureCoding {
...
 		[MacCatalyst (14,0)]
 		[NullAllowed]
 		[Export ("animationMetrics", ArgumentSemantic.Strong)]
 		MXAnimationMetric AnimationMetrics { get; }
 
-		[iOS (14,0)]
+		[NoMac, iOS (14,0)]
```
Comment (spouliot) [inline src/metrickit.cs:399]: minor: not needed, there's a `[NoMac]` on the type-level

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Unrelated issue: https://github.com/xamarin/maccore/issues/2414

**Pull request merged**